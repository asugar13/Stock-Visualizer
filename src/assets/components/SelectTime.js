import React from "react";
import Select from "react-select"

export default ({ setTime }) => {

  const timeOptions = [{ value: 1000, label: "1 second" }, { value: 2000, label: "2 seconds" }, { value: 3000, label: "3 seconds" }, { value: 4000, label: "4 seconds" }, { value: 5000, label: "5 seconds" }]
  const changeTime = (option) => {
    setTime(option.value)
  }
  return (
    <div className="select">
      <Select
        options={timeOptions}
        onChange={changeTime}
        placeholder="Set the interval (default 1s)"
      />
    </div>
  )
}