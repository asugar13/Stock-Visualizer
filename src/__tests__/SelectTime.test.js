import React from 'react';
import renderer from 'react-test-renderer';
import SelectTime from '../assets/components/SelectTime';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("<SelectTime /> with no props", () => {
  const container = shallow(<SelectTime />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })

  it('should have the class select', () => {
    expect(container.find('div[className="select"]').length).toEqual(1);
  });
})