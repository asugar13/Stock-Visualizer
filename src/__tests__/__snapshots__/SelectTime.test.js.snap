// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectTime /> with no props should match the snapshot 1`] = `"<div class=\\"select\\"><div class=\\" css-2b097c-container\\"><span aria-live=\\"polite\\" aria-atomic=\\"false\\" aria-relevant=\\"additions text\\" class=\\"css-1f43avz-a11yText-A11yText\\"></span><div class=\\" css-yk16xz-control\\"><div class=\\" css-g1d714-ValueContainer\\"><div class=\\" css-1wa3eu0-placeholder\\">Set the interval (default 1s)</div><div class=\\"css-b8ldur-Input\\"><div class=\\"\\" style=\\"display:inline-block\\"><input type=\\"text\\" autoCapitalize=\\"none\\" autoComplete=\\"off\\" autoCorrect=\\"off\\" id=\\"react-select-2-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" value=\\"\\" aria-autocomplete=\\"list\\" style=\\"box-sizing:content-box;width:1px;label:input;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit\\"/><div style=\\"position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre\\"></div></div></div></div><div class=\\" css-1hb7zxy-IndicatorsContainer\\"><span class=\\" css-1okebmr-indicatorSeparator\\"></span><div class=\\" css-tlfecz-indicatorContainer\\" aria-hidden=\\"true\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-tj5bde-Svg\\"><path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path></svg></div></div></div></div></div>"`;
